src/types/simple.rs

```diff
@@ -478,121 +457,19 @@ impl TypeRow<ClassicType> {
     }
 }
 
-// TODO some of these, but not all, will probably want exposing via
-// pyo3 wrappers eventually.
 impl<T: PrimType> TypeRow<T> {
-    /// Create a new empty row.
-    pub const fn new() -> Self {
-        Self {
-            types: Cow::Owned(Vec::new()),
-        }
-    }
-
-    /// Iterator over the types in the row.
-    pub fn iter(&self) -> impl Iterator<Item = &T> {
-        self.types.iter()
-    }
-
-    /// Returns the number of types in the row.
-    #[inline(always)]
-    pub fn len(&self) -> usize {
-        self.types.len()
-    }
-
-    /// Returns `true` if the row contains no types.
-    #[inline(always)]
-    pub fn is_empty(&self) -> bool {
-        self.types.len() == 0
-    }
-
     /// Returns whether the row contains only hashable classic data.
     #[inline(always)]
     pub fn purely_hashable(&self) -> bool {
```

Comment:
Why isn't this one (and `containing_tag`) in `type_row.rs`?

Reply:
Because it only applies to TypeRow's of PrimType, not (for example) to TypeRow's of SerSimpleType